### PR TITLE
general refactors and adding functionality to vote for reviews

### PR DESCRIPTION
**Pull request by SZahl**

### CODE REVIEW

File: src/utils/api.js [hunk 1]
```diff
@@ -28,4 +27,18 @@ export const fetchComments = (review_id) => {
...
+    const increment = { inc_votes: 1 };
+    return reviews
+    .patch(`/reviews/${review_id}`, increment)
+    .then(({ data }) => {
+        setCurrentVotes(data.votes);
+        setButtonDisable(true);
+        setVoteCastText('Your vote has been added!')
+    })
+    .catch((error) => {
+        setVoteCastText('Error, please try again!')
```
Comment (popatre) [inline src/utils/api.js]: Your api file shouldnt handle the setting of state, you should just use this to complete the request to the db. 

Any state setting should happen in the react component. 

You also need to update the votes optimisitically. At the moment, on a slow connection, I click the button and nothing happens for a while. I could then click it multiple times, thinking its not working and end up doing multiple requests

File: src/Components/DetailedReview.jsx [hunk 2]
```diff
@@ -36,7 +36,9 @@ if (isLoading) {
             <p>{currentReviewByID.review_body}</p>
             <p>Category :- {currentReviewByID.category}</p>
             <p>Posted at {currentReviewByID.created_at}</p>
-            <p>Current votes :- {currentReviewByID.votes}</p>
+            <p>Current votes :- {currentVotes}</p>
+            <button className="voteButton" disabled={buttonDisable} onClick={() => {addVotes(review_id, setCurrentVotes, setButtonDisable, setVoteCastText)}}>Vote for this review!</button><br />
```
Comment (popatre) [inline src/Components/DetailedReview.jsx]: Please add your functionality for updating the votes in this component instead, please

**Review by popatre — CHANGES_REQUESTED** [review]
Your api file shouldnt handle the setting of state, you should just use this to complete the request to the db. 

Any state setting should happen in the react component. 

You also need to update the votes optimisitically. At the moment, on a slow connection, I click the button and nothing happens for a while. I could then click it multiple times, thinking its not working and end up doing multiple requests

**Review by aloush-dev — APPROVED** [review]
Looking good, I would take currentVotes out of your dependancy array as currently its causing a full rerender of the page when you use the vote button

**Pull request merged**